packages/adblocker/src/engine/bucket/cosmetic.ts

```diff
@@ -397,6 +397,8 @@ export default class CosmeticFilterBucket {
     getExtendedRules = true,
     getRulesFromDOM = true,
     getRulesFromHostname = true,
+
+    isFilterEligible = () => true,
```

Comment:
I would instead default to `undefined` and only evaluate `isFilterEligible` if it's specified. No need to create and evaluate a function if we disable the preprocessors.

Reply:
True, I'll remove that default value and check if `isFilterEligible` is given in the matching function.

Reply:
Updated to check if `isFilterExcluded` exists.